### ISSUE TITLE
ESM import/export specifier spacing

### ISSUE THREAD

**Issue by demurgos** [eslint/eslint#13948]
**Please describe what the rule should do:**

Add a rule to enforce consistent spacing in ESM import and export specifiers such as `import {foo}` from "./foo.js" or `export { foo }`.

This rule would be the ESM specifier analog of [`object-curly-spacing`](https://eslint.org/docs/rules/object-curly-spacing).

**What new ECMAScript feature does this rule relate to?**

No new feature, ES module exist since 2015.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[x] Other: layout, style

**Provide 2-3 code examples that this rule will warn about:**

```js
// Warns if the rule is `always`
import {foo} from "foo";
// Expected:
import { foo } from "foo";

// Warns if the rule is `never`
export { foo, bar as baz };
// Expected:
export {foo, bar as baz};

// Warns if the rule is `never`
import * as foo, { bar, baz } from "foo";
// Expected:
import * as foo, {bar, baz} from "foo";
```

**Why should this rule be included in ESLint (instead of a plugin)?**

ES modules are a core part of Javascript, enforcing consistent styling around them is necessary for any team wishing to enforce a style guide. I tried to find an existing rule or plugin to solve this issue but couldn't find one.

**Are you willing to submit a pull request to implement this rule?**

No. I am not familiar enough with ESLint and do not have enough time to dig into it currently. I may eventually looking into submitting the PR, but probably not soon.

**Comment by mdjermanovic** [comment]
Hi @demurgos, thanks for the issue!

`object-curly-spacing` already works with `{}` in imports and exports, as in [this demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG9iamVjdC1jdXJseS1zcGFjaW5nOiBbXCJlcnJvclwiLCBcImFsd2F5c1wiXSAqL1xuXG5pbXBvcnQge2F9IGZyb20gXCJmb29cIjtcblxuaW1wb3J0IGIsIHtjLCBkfSBmcm9tIFwiZm9vXCI7XG5cbmV4cG9ydCB7YX07XG5cbmV4cG9ydCB7ZX0gZnJvbSBcImZvb1wiOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==).

**Comment by demurgos** [comment]
Sorry for opening the issue: I feel silly, there is even an example in the link I provided for `object-curly-spacing`.
